modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/order/term/contributor/CommerceOrderCommerceDefinitionTermContributor.java

```diff
@@ -119,6 +232,193 @@ public String getFilledTerm(String term, Object object, Locale locale)
 				commerceOrder.getShippingAddress(), locale);
 		}
 
+		if (term.equals(_ORDER_SHIPPING_OPTION)) {
+			CommerceShippingFixedOption commerceShippingFixedOption =
+				_commerceShippingFixedOptionLocalService.
+					fetchCommerceShippingFixedOption(
+						commerceOrder.getCompanyId(),
+						commerceOrder.getShippingOptionName());
+
+			if (commerceShippingFixedOption != null) {
+				return commerceShippingFixedOption.getName(locale);
+			}
+
+			CommerceShippingMethod commerceShippingMethod =
+				commerceOrder.getCommerceShippingMethod();
+
+			if (commerceShippingMethod != null) {
+				CommerceShippingEngine commerceShippingEngine =
+					_commerceShippingEngineRegistry.getCommerceShippingEngine(
+						commerceShippingMethod.getEngineKey());
+
+				if (commerceShippingEngine != null) {
+					List<CommerceShippingOption> commerceShippingOptions =
+						commerceShippingEngine.getCommerceShippingOptions(
+							null, commerceOrder, locale);
+
+					Stream<CommerceShippingOption>
+						commerceShippingOptionsStream =
+							commerceShippingOptions.stream();
+
+					CommerceShippingOption commerceShippingOption =
+						commerceShippingOptionsStream.filter(
+							cso -> cso.getKey(
+							).equals(
+								commerceOrder.getShippingOptionName()
+							)
+						).findFirst(
+						).orElse(
+							null
+						);
+
+					if (commerceShippingOption != null) {
+						return commerceShippingOption.getName();
+					}
+
+					return "";
+				}
+			}
+
+			return "";
+		}
+
+		if (term.equals(_ORDER_SHIPPING_TOTAL)) {
+			BigDecimal shippingAmount = commerceOrder.getShippingAmount();
+
+			if (shippingAmount != null) {
+				CommerceCurrency commerceCurrency =
+					commerceOrder.getCommerceCurrency();
+
+				BigDecimal shippingAmountRound = commerceCurrency.round(
+					shippingAmount);
+
+				return shippingAmountRound.toPlainString();
+			}
+
+			return "";
+		}
+
+		if (term.equals(_ORDER_SHIPPING_WITH_TAX_TOTAL)) {
+			BigDecimal shippingWithTaxAmount =
+				commerceOrder.getShippingWithTaxAmount();
+
+			if (shippingWithTaxAmount != null) {
+				CommerceCurrency commerceCurrency =
+					commerceOrder.getCommerceCurrency();
+
+				BigDecimal shippingWithTaxAmountRound = commerceCurrency.round(
+					shippingWithTaxAmount);
+
+				return shippingWithTaxAmountRound.toPlainString();
+			}
+
+			return "";
+		}
+
+		if (term.equals(_ORDER_TAX_TOTAL)) {
+			BigDecimal taxAmount = commerceOrder.getTaxAmount();
+
+			if (taxAmount != null) {
+				CommerceCurrency commerceCurrency =
+					commerceOrder.getCommerceCurrency();
+
+				BigDecimal taxAmountRound = commerceCurrency.round(taxAmount);
+
+				return taxAmountRound.toPlainString();
+			}
+
+			return "";
+		}
+
+		if (term.equals(_ORDER_TOTAL)) {
+			BigDecimal total = commerceOrder.getTotal();
+
+			if (total != null) {
+				CommerceCurrency commerceCurrency =
+					commerceOrder.getCommerceCurrency();
+
+				return String.valueOf(commerceCurrency.round(total));
```

Comment:
Why are we using 2 different method of retrieving this value compared to _ORDER_TAX_TOTAL?
I see String.valueOf(commerceCurrency.round(total)) in the first and taxAmountRound.toPlainString() in the latter one.